src/mlpack/methods/ann/layer/log_softmax_impl.hpp

```diff
@@ -97,6 +99,22 @@ void LogSoftMaxType<MatType>::Forward(const MatType& input, MatType& output)
   output = input - maxInput;
 }
 
+#ifdef MLPACK_HAS_COOT
+
+template<typename MatType>
+void LogSoftMaxType<MatType>::Forward(const MatType& input, MatType& output,
+                                      const typename std::enable_if_t<
+                                      coot::is_coot_type<MatType>::value>*)
+{
+  MatType maxInput = repmat(max(input), input.n_rows, 1);
+  output = (maxInput - input);
+  output = exp(output * -1);
+  maxInput.each_row() += log(sum(output));
```

Comment:
technically, we still do not have an `each_row()` function in bandicoot

Reply:
Yes, it is possible and on my todo list though.  (Soon enough... :smile:)